### PR TITLE
feat: adding custom panel location capability

### PR DESCRIPTION
**Pull request by halenstalians23**
- **Please check if the PR fulfills these requirements**

* [x] The commit message follows our guidelines
* [ ] Tests for the changes have been added (for bug fixes/features)
* [ ] Docs have been added / updated (for bug fixes / features)

- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- Allows capability of specifying panel location for each individual panel type (errors, workflow parameters, etc.) This is needed for porting the errors pane into PPUX, as the spec requires it to render on a different side of the designer than in LAUX. This change lays the foundation, existing behavior is unchanged.

- **What is the current behavior?** (You can also link to an open issue here)
- All panels display on right side of designer.

- **What is the new behavior (if this is a feature change)?**
- Same as current behavior.

- **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
N/A

- **Please Include Screenshots or Videos of the intended change**:
N/A

AB#24899016

### CODE REVIEW

File: libs/designer/src/lib/ui/panel/panelRoot.tsx [hunk 1]
```diff
@@ -61,6 +61,13 @@ export const PanelRoot = (props: PanelRootProps): JSX.Element => {
     [currentPanelMode]
   );
 
+  const customPanelLocations: CustomPanelLocation[] = [];
+  for (const customPanelLocation of customPanelLocations) {
+    if (currentPanelMode === customPanelLocation.panelMode) {
+      commonPanelProps.panelLocation = customPanelLocation.panelLocation;
```
Comment (hartra344) [inline libs/designer/src/lib/ui/panel/panelRoot.tsx:67]: please move this into a local variable instead of changing the props themselves.

**Comment by halenstalians23** [comment]
@microsoft-github-policy-service agree company="Microsoft"

File: libs/designer/src/lib/ui/panel/panelRoot.tsx [hunk 2]
```diff
@@ -61,6 +61,13 @@ export const PanelRoot = (props: PanelRootProps): JSX.Element => {
     [currentPanelMode]
   );
 
+  const customPanelLocations: CustomPanelLocation[] = [];
+  for (const customPanelLocation of customPanelLocations) {
```
Comment (hartra344) [inline libs/designer/src/lib/ui/panel/panelRoot.tsx:65]: you're initializing customPanelLocations to an empty array above so this will never get ran?

Comment (halenstalians23) [reply libs/designer/src/lib/ui/panel/panelRoot.tsx:65]: I will be passing this in as a Designer prop, this was only temporary, will update in next commit

File: libs/designer/src/lib/ui/panel/panelRoot.tsx (hunk 1 above)
Comment (halenstalians23) [reply libs/designer/src/lib/ui/panel/panelRoot.tsx:67]: ok, putting in memo function in next commit

File: libs/designer/src/lib/ui/panel/panelRoot.tsx [hunk 3]
```diff
@@ -45,16 +46,21 @@ export const PanelRoot = (props: PanelRootProps): JSX.Element => {
 
   const dismissPanel = useCallback(() => dispatch(clearPanel()), [dispatch]);
 
-  const commonPanelProps: CommonPanelProps = useMemo(
-    () => ({
+  const commonPanelProps: CommonPanelProps = useMemo(() => {
+    let customLocation = undefined;
```
Comment (hartra344) [inline libs/designer/src/lib/ui/panel/panelRoot.tsx:50]:
```suggestion
    let customLocation = customPanelLocations?.find(x => currentPanelMode === x.panelMode);
```

then you can remove the entire loop below

**Pull request closed without merging**